utils/src/main/java/io/confluent/common/utils/TestUtils.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.common.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+/**
+ * Utility functions to be used in unit/integration tests
+ */
+public class TestUtils {
+
+  /**
+   * Create a temporary directory. The directory and any contents will be deleted when the test
+   * process terminates.
+   */
+  public static File tempDirectory() {
+    final File file;
+    try {
+      file = Files.createTempDirectory("confluent").toFile();
```

Comment:
Is there a guarantee that a temp directory will always be on the default filesystem?

Reply:
@trejkaz What do you mean by "default" filesystem? The same as `/` or are you thinking in terms of a non-posix platform?

Short version is no, there shouldn't be any such assumption here. For example, `/tmp` could easily be on a standalone, memory-backed disk.

Reply:
"default" in the sense of `FileSystems.getDefault()`, and the note on `Path.toFile()` saying that it will throw `UnsupportedOperationException` if it isn't on the default one.

But I just re-read the docs on `createTempDirectory`, and indeed, if you don't use the overload taking `Path`, the file it creates _is_ created on the default filesystem. Which is good, this is another assumption that can be used to permit some calls to toFile(). (I'm in the middle of a several year war to stomp them out in our own project.)